File: client/src/app/drive.ts
import { ReasonKey } from './reason';
import { DriveTemplate } from './drive-template';

export interface Drive {
  id?: string | null;
  template: DriveTemplate | null;
  date: Date;
  reason?: ReasonKey | null;
  fromLocation?: string | null;
  toLocation?: string | null;
  driveLength?: number | null;
}

export interface DriveRequest {
  id?: string | null;
  templateId: string | null;
  date: string;
  reason?: ReasonKey | null;
  fromLocation?: string | null;
  toLocation?: string | null;
  driveLength?: number | null;
}

File: src/main/java/de/schrell/drives/drives/domain/entities/Drive.java
package de.schrell.drives.drives.domain.entities;

import jakarta.annotation.Nullable;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString(doNotUseGetters = true)
@Entity
public class Drive {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;


    @ManyToOne
    @JoinColumn(name = "template_id")
    @Nullable
    private DriveTemplate template;

    private LocalDate date;

    @Nullable
    private Reason reason;

    @Nullable
    @Column(name = "from_location")
    private String fromLocation;

    @Nullable
    @Column(name = "to_location")
    private String toLocation;

    @Nullable
    @Column(name = "drive_length")
    private Integer driveLength;
}
